lib/voxpupuli/release/rake_tasks.rb

```diff
@@ -52,3 +52,90 @@
     fail "Unable to find a CHANGELOG.md entry for the #{v} release."
   end
 end
+
+desc "Prepare a release"
+task "release:prepare" do
+  Rake::Task["release:porcelain:changelog"].invoke
+  Rake::Task["strings:generate:reference"].invoke if File.exist?('REFERENCE.md')
+end
+
+begin
+  require 'github_changelog_generator/task'
+  require 'puppet_blacksmith'
+
+  class GCGConfig
+    def self.user=(user)
+      @user = user
+    end
+
+    def self.user
+      @user || metadata['name'].split(%r{[-/]}).first
+    end
+
+    def self.metadata
+      @metadata ||= Blacksmith::Modulefile.new.metadata
+    end
+  end
+
+  task "release:porcelain:changelog" do
+    # This is taken from lib/github_changelog_generator/task
+    # The generator cannot be used because we want to lazyly evaluate
+    # GCGConfig.user which might be overrider in the module Rakefile.
+    options = GitHubChangelogGenerator::Parser.default_options
+    options[:user] = GCGConfig.user
+    options[:project] = GCGConfig.metadata['name']
+    options[:future_release] = "v#{GCGConfig.metadata['version']}" if GCGConfig.metadata['version'] =~ /^\d+\.\d+.\d+$/
+    options[:header] = <<~HEADER.chomp
+      # Changelog
+
+      All notable changes to this project will be documented in this file.
+      Each new release typically also includes the latest modulesync defaults.
+      These should not affect the functionality of the module.
+    HEADER
+    options[:exclude_labels] = %w{duplicate question invalid wontfix wont-fix modulesync skip-changelog}
+    generator = GitHubChangelogGenerator::Generator.new(options)
+    log = generator.compound_changelog
+    output_filename = options[:output].to_s
+    File.write(output_filename, log)
+    puts "Done!"
+    puts "Generated log placed in #{File.absolute_path(output_filename)}"
+  end
+
+  # Workaround for https://github.com/github-changelog-generator/github-changelog-generator/issues/715
+  require 'rbconfig'
+  if RbConfig::CONFIG['host_os'] =~ /linux/
+    task "release:porcelain:changelog" do
+      puts 'Fixing line endings...'
+      changelog_file = 'CHANGELOG.md'
+      changelog_txt = File.read(changelog_file)
+      new_contents = changelog_txt.gsub(%r{\r\n}, "\n")
+      File.write(changelog_file, new_contents)
+    end
+  end
+rescue LoadError, Blacksmith::Error
```

Comment:
I'd prefer to have a minimal block with rescue. So ideally:
```ruby
begin
  require '...'
rescue LoadError
  task "release:porcelain:changelog" do
    fail 'Missing github_changelog_generator gem'
  end
else
  # ...
end
```

And then catch `Blacksmith::Error` locally and make it fail.

Reply:
Done. I got rid of the `Blacksmith::Error` catching: when metadata.json is absent we are probably not ready for rolling a release, and the error is quite descriptive:

```
Blacksmith::Error: Unable to find any of ["metadata.json"]
```